src/System.Linq.Expressions/tests/Convert/ConvertTests.cs

```diff
@@ -16710,5 +16710,103 @@ public static void PointerType()
         {
             Assert.Throws<ArgumentException>("type", () => Expression.Convert(Expression.Constant(null), typeof(int*)));
         }
+
+        public static IEnumerable<object[]> Conversions()
+        {
+            yield return new object[] { 3, 3 };
+            yield return new object[] { (byte)3, 3 };
+            yield return new object[] { 3, 3.0 };
+            yield return new object[] { 3.0, 3 };
+            yield return new object[] { 12345678, (short)24910 };
```

Comment:
12345678 should cast to 24910  checked and throw unchecked, so the behaviour should be different. The flip-side of making sure it does indeed throw I think is covered well enough by the existing tests.